### PR TITLE
Several improvements and generalization

### PR DESCRIPTION
**Pull request by gciatto**
Breaking changes for @DanySK:
- [3f37e90c5b4c5264e50a78cc7ff2e2d4451fb666](https://github.com/DanySK/build-check-deploy-gradle-action/commit/3f37e90c5b4c5264e50a78cc7ff2e2d4451fb666)
- [03f7f20ac033f6f15e54b50fed3f7abd0084676a](https://github.com/DanySK/build-check-deploy-gradle-action/commit/03f7f20ac033f6f15e54b50fed3f7abd0084676a)

### CODE REVIEW

**Review by DanySK — COMMENTED** [review]
This is plain wrong.

File: action.yml [hunk 1]
```diff
@@ -5,13 +5,17 @@ branding:
...
 
 inputs:
+  pre-build-command:
+    description: 'A command to be executed before the build phase (but after configuring the JVM)'
+    default: 'true'
+    required: false
   build-command:
     description: 'The command executed in the build phase'
-    default: './gradlew assemble --parallel'
+    default: 'gradle assemble --parallel'
```
Comment (DanySK) [inline action.yml:14]: revert this, the wrapper **must** always be used.

File: action.yml [hunk 2]
```diff
@@ -156,20 +181,30 @@ runs:
...
+        SIGNING_PASSWORD: ${{ inputs.signing-password }}
+        #
+        ORG_GRADLE_PROJECT_mavenRepo: ${{ inputs.maven-central-repo }}
+        ORG_GRADLE_PROJECT_mavenUsername: ${{ inputs.maven-central-password }}
+        ORG_GRADLE_PROJECT_mavenPassword: ${{ inputs.maven-central-username }}  
         ORG_GRADLE_PROJECT_signingKey: ${{ inputs.signing-key }}
         ORG_GRADLE_PROJECT_signingPassword: ${{ inputs.signing-password }}
-        SIGNING_PASSWORD: ${{ inputs.signing-password }}
+        ORG_GRADLE_PROJECT_npmRepo: ${{ inputs.npm-repo }}
+        ORG_GRADLE_PROJECT_npmToken: ${{ inputs.npm-token }}
```
Comment (DanySK) [inline action.yml:192]: These are nice. I would also expose them as `NPM_REPO` and `NPM_TOKEN`, consistently with how we deal with all other variables

File: action.yml [hunk 3]
```diff
@@ -156,20 +181,30 @@ runs:
         GRADLE_PUBLISH_KEY: ${{ inputs.gradle-publish-key }}
         MAVEN_CENTRAL_PASSWORD: ${{ inputs.maven-central-password }}
         MAVEN_CENTRAL_USERNAME: ${{ inputs.maven-central-username }}
+        SIGNING_PASSWORD: ${{ inputs.signing-password }}
+        #
```
Comment (DanySK) [inline action.yml:185]:
```suggestion
        # Gradle-specific exports, see: 
        # https://docs.gradle.org/current/userguide/build_environment.html#sec:project_properties
```

File: action.yml [hunk 4]
```diff
@@ -156,20 +181,30 @@ runs:
...
       shell: bash
       run: |
+        echo "::group::Cleanup"
         ${{ inputs.clean-command }}
+        echo "::endgroup::"
     - name: Turn off the Gradle Daemon
       shell: bash
       working-directory: ${{ inputs.working-directory }}
-      run: ./gradlew --stop
+      run: gradle --stop
```
Comment (DanySK) [inline action.yml:207]: to be reverted along with all the calls to unknown versions of gradle

**Review by DanySK — CHANGES_REQUESTED** [review]
A couple of changes to reify, but several ideas look very good.

File: action.yml (hunk 3 above)
Comment (gciatto) [reply action.yml:187]: Should I do something here?

**Pull request merged**